cpp/src/io/parquet/parquet.hpp

```diff
@@ -495,6 +646,53 @@ ParquetFieldStructFunctor<T> ParquetFieldStruct(int f, T &v)
   return ParquetFieldStructFunctor<T>(f, v);
 }
 
+/**
+ * @brief Functor to read a union member from CompactProtocolReader
+ *
+ * @return True if field types mismatch or if the process of reading a
+ * union member fails
+ */
+template <typename T, bool is_empty = false>
```

Comment:
Could you add a `tparam` documentation here talking about `is_empty` is meant to be set to true if `T` is an empty type.